week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,12 +1,38 @@
 // Implement the missing functionality according to the task description.
 // Hint: Remember which function you need to call when it's time to alarm.
+let intervalID;
+let timeRemainingdisplay = document.querySelector('#timeRemaining');
+let timeInput = document.querySelector('#alarmSet');
+let setAlarmButton = document.querySelector('#set');
+setAlarmButton.addEventListener('click', setAlarm);
 
-function setAlarm() { }
+
+function setAlarm() { 
+  intervalID = setInterval (function () {
+    if(timeInput.value > 0){
+    timeRemainingdisplay.textContent = `Time remaining: 00:0${timeInput.value}` 
+    timeInput.value --;
+    
+  } else if (timeInput.value <= 0) {
+    timeRemainingdisplay.textContent = `Time remaining: 00:00`;
+    playAlarm();
+    stopNoise();
+  }
+  
+}, 1000);
+}
+  
+function stopNoise () {
+  clearInterval(intervalID);
+}
```

Comment:
Your function is correct, what your missing inside stopNoise function is audio.pause() when you add that the noise will stop. Work on the seconds  (remove the extra zero ).